classes/Http/Action/Talk/UpdateAction.php

```diff
@@ -0,0 +1,236 @@
+
+    /**
+     * @var View\TalkHelper
+     */
+    private $talkHelper;
+
+    /**
+     * @var CallForPapers
+     */
+    private $callForPapers;
+
+    /**
+     * @var HTMLPurifier
+     */
+    private $purifier;
+
+    /**
+     * @var Swift_Mailer
+     */
+    private $swiftMailer;
+
+    /**
+     * @var string
+     */
+    private $applicationEmail;
+
+    /**
+     * @var string
+     */
+    private $applicationTitle;
+
+    /**
+     * @var string
+     */
+    private $applicationEndDate;
+
+    /**
+     * @var Twig_Environment
+     */
+    private $twig;
+
+    /**
+     * @var Routing\Generator\UrlGeneratorInterface
+     */
+    private $urlGenerator;
+
+    public function __construct(
+        Services\Authentication $authentication,
+        View\TalkHelper $talkHelper,
+        CallForPapers $callForPapers,
+        HTMLPurifier $purifier,
+        Swift_Mailer $swiftMailer,
+        Twig_Environment $twig,
+        Routing\Generator\UrlGeneratorInterface $urlGenerator,
+        string $applicationEmail,
+        string $applicationTitle,
+        string $applicationEndDate
+    ) {
+        $this->authentication     = $authentication;
+        $this->talkHelper         = $talkHelper;
+        $this->callForPapers      = $callForPapers;
+        $this->purifier           = $purifier;
+        $this->swiftMailer        = $swiftMailer;
+        $this->twig               = $twig;
+        $this->urlGenerator       = $urlGenerator;
+        $this->applicationEmail   = $applicationEmail;
+        $this->applicationTitle   = $applicationTitle;
+        $this->applicationEndDate = $applicationEndDate;
+    }
+
+    public function __invoke(HttpFoundation\Request $request): HttpFoundation\Response
+    {
+        if (!$this->callForPapers->isOpen()) {
+            $request->getSession()->set('flash', [
+                'type'  => 'error',
+                'short' => 'Read Only',
+                'ext'   => 'You cannot edit talks once the call for papers has ended',
+            ]);
+
+            $url = $this->urlGenerator->generate('talk_view', [
+                'id' => $request->get('id'),
+            ]);
+
+            return new HttpFoundation\RedirectResponse($url);
+        }
+
+        $user = $this->authentication->user();
+
+        $requestData = [
+            'id'          => $request->get('id'),
+            'title'       => $request->get('title'),
+            'description' => $request->get('description'),
+            'type'        => $request->get('type'),
+            'level'       => $request->get('level'),
+            'category'    => $request->get('category'),
+            'desired'     => $request->get('desired'),
+            'slides'      => $request->get('slides'),
+            'other'       => $request->get('other'),
+            'sponsor'     => $request->get('sponsor'),
+            'user_id'     => $request->get('user_id'),
+        ];
+
+        $form = $this->createTalkForm($requestData);
```

Comment:
Since the form isn't injected, but created here, we can only write unit tests up until the condition above.